### ISSUE TITLE
Scaffold.bottomSheet is toggled by setState

### ISSUE THREAD

**Issue by student020341** [flutter/flutter#68522]
This might just be my not understanding something, but it doesn't look right to me. The bottomSheet isn't using any state variables so I'm not sure why it's opening and closing when a state var is incremented. 

## Steps to Reproduce

I managed to reduce this issue to ~30 lines of code to reproduce the issue

```
import 'package:flutter/material.dart';

class Test extends StatefulWidget {
  @override
  _Test createState() => _Test();
}

class _Test extends State<Test> {
  int count = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(title: Text("appbar")),
        body: RaisedButton(
          child: Text("count $count"),
          onPressed: () {
            setState(() {
              count = count + 1;
            });
          },
        ),
        bottomSheet: Container(
          color: Colors.green,
          height: 300,
        ));
  }
}
```

**Expected results:** 

I expected a permanently displayed bottom sheet

**Actual results:**

When setState fires, the bottom sheet animates closed if it is open and pops up if it's closed. 

There are no exceptions logged and this issue is encountered running on android or web. This may or may not be related to issue #43485

One last detail from playing around - I can get around this issue by referencing a widget declared as final, but that won't work for my purposes because I do want to eventually control the content of the bottom sheet with state.

**Comment by student020341** [comment]
Another detail! It looks like this is only happening as a route. This main code may also be needed to reproduce:

```
void main() {
  runApp(MaterialApp(
    theme: ThemeData(primaryColor: Colors.white),
    debugShowCheckedModeBanner: false,
    title: 'demo',
    initialRoute: '/test',
    routes: {
      "/test": (context) => Test()
    },
  ));
}
```

**Comment by TahaTesser** [comment]
Hi @student020341 
It doesn't close for me on the latest `stable` channel on iOS or Android

![ezgif com-gif-maker](https://user-images.githubusercontent.com/48603081/96563432-931b9f00-12df-11eb-8522-7e5c5798efec.gif)



<details> 
<summary>complete code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Test(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

class Test extends StatefulWidget {
  @override
  _Test createState() => _Test();
}

class _Test extends State<Test> {
  int count = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(title: Text("appbar")),
        body: RaisedButton(
          child: Text("count $count"),
          onPressed: () {
            setState(() {
              count = count + 1;
            });
          },
        ),
        bottomSheet: Container(
          color: Colors.green,
          height: 300,
        ));
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.2, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.22.2 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 84f3d28555 (4 days ago), 2020-10-15 16:26:19 -0700
    • Engine revision b8752bbfff
    • Dart version 2.10.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (2 available)
    • RMX2001 (mobile)   • EUYTFEUSQSRGDA6D                     • android-arm64
      • Android 10 (API 29)
    • iPhone 11 (mobile) • 193EBBC9-CC2A-4477-B8A0-A5F12BB51400 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

! Doctor found issues in 1 category.

```

</details> 

Can you please provide your `flutter doctor -v` your `flutter run --verbose` and a complete reproducible minimal code sample
Thank you

**Comment by student020341** [comment]
Ah, I forgot I was on a beta channel from a few weeks ago while checking out routing 2.0. I just tried to make a new repo to demonstrate the issue, ran flutter upgrade, and the issue is gone from my original project now. I don't know what version I was on, sorry!

The issue appears to be resolved for the current beta and I should probably switch back to a stable channel. Thanks anyway. 

While I was having the issue I noticed if I made the bottom sheet **final** it would not show the issue. I also didn't see the issue if I ran it directly instead of using a route, some combination of those factors were the issue for that specific beta version, I guess.
